### ISSUE TITLE
`StackOverflowError` when `min_level=Logging.Debug`

### ISSUE THREAD

**Issue by JJMinton** [PhilipVinc/TensorBoardLogger.jl#96]
As the title explains, I'm running 
```
using TensorBoardLogger
using Logging

tb_logger = TBLogger("tensorboard_logs/run", min_level=Logging.Debug)
with_logger(tb_logger) do
    @info "loss" loss=10.
end
```
and get
```
ERROR: LoadError: StackOverflowError:
Stacktrace:
 [1] fieldnames at ./reflection.jl:173 [inlined]
 [2] propertynames at ./reflection.jl:1371 [inlined]
 [3] logable_propertynames at /root/.julia/packages/TensorBoardLogger/tMpIk/src/logger_dispatch.jl:39 [inlined]
 [4] preprocess(::String, ::Type{TensorBoardLogger.tensorboard.AllocationDescription}, ::Array{Pair{String,Any},1}) at /root/.julia/packages/TensorBoardLogger/tMpIk/src/logger_dispatch.jl:16
 [5] preprocess(::String, ::Module, ::Array{Pair{String,Any},1}) at /root/.julia/packages/TensorBoardLogger/tMpIk/src/logger_dispatch.jl:19 (repeats 20113 times)
 [6] handle_message(::TBLogger, ::Base.CoreLogging.LogLevel, ::String, ::Module, ::Symbol, ::Symbol, ::String, ::Int64; kwargs::Base.Iterators.Pairs{Symbol,ProtoBuf.ProtoMeta,Tuple{Symbol},NamedTuple{(:meta,),Tuple{ProtoBuf.ProtoMeta}}}) at /root/.julia/packages/TensorBoardLogger/tMpIk/src/TBLogger.jl:241
 [7] macro expansion at ./logging.jl:332 [inlined]
 [8] writeproto(::Base.GenericIOBuffer{Array{UInt8,1}}, ::TensorBoardLogger.tensorboard.Event, ::ProtoBuf.ProtoMeta) at /root/.julia/packages/ProtoBuf/NJbMU/src/codec.jl:420
 [9] writeproto at /root/.julia/packages/ProtoBuf/NJbMU/src/codec.jl:419 [inlined]
 [10] write_event(::IOStream, ::TensorBoardLogger.tensorboard.Event) at /root/.julia/packages/TensorBoardLogger/tMpIk/src/event.jl:31
 [11] handle_message(::TBLogger, ::Base.CoreLogging.LogLevel, ::String, ::Module, ::Symbol, ::Symbol, ::String, ::Int64; kwargs::Base.Iterators.Pairs{Symbol,Float64,Tuple{Symbol},NamedTuple{(:loss,),Tuple{Float64}}}) at /root/.julia/packages/TensorBoardLogger/tMpIk/src/TBLogger.jl:250
 [12] macro expansion at ./logging.jl:332 [inlined]
 [13] (::var"#1#2")() at /home/jupyter/project/scripts/train.jl:6
 [14] with_logstate(::Function, ::Any) at ./logging.jl:408
 [15] with_logger(::Function, ::TBLogger) at ./logging.jl:514
 [16] top-level scope at /home/jupyter/project/scripts/train.jl:11
 [17] include(::Function, ::Module, ::String) at ./Base.jl:380
 [18] include(::Module, ::String) at ./Base.jl:368
 [19] exec_options(::Base.JLOptions) at ./client.jl:296
 [20] _start() at ./client.jl:506
in expression starting at /home/jupyter/project/scripts/train.jl:5
```

And my environment:
```
(@v1.5) pkg> status
Status `~/.julia/environments/v1.5/Project.toml`
  [7073ff75] IJulia v1.23.2
  [899adc3e] TensorBoardLogger v0.1.15
  [cf7118a7] UUIDs
```

**Comment by PhilipVinc** [comment]
Oh, wow. 
That is because when you set `with_logger(tbl)` all the `@debug` messages will be routed to TensorBoardLogger, which does a bunch of things and eventually calls `ProtoBuf.writeproto(...)` to serialise that data to tensor board format.

However,[`ProtoBuf.writeproto`](https://github.com/JuliaIO/ProtoBuf.jl/blob/36a5ae8ed44156b062b7157ff6562306a7200207/src/codec.jl#L418) has some debug statements itself, that invoke TensorBoardLogger logic, that invoke `writeproto` in an infinite recursion.

@oxinabox and @c42f I am not sure what would be the best way to solve this. You Logging gurus have an idea?

**Comment by PhilipVinc** [comment]
I think in general we should disable TensorBoardLogger itself as a logger from function calls called from within TensorBoardLogger. 
I'm unsure on how to do that.

**Comment by JJMinton** [comment]
Thanks for the quick response. I'm happy to help anyway I can, but still a bit new to Julia.

**Comment by PhilipVinc** [comment]
Bump @oxinabox @c42f I don't know how to solve this issue and I'd gladly accept a pointer

**Comment by oxinabox** [comment]
We should probably wrap our calls to `writeproto` in
```julia
withlogger(NullLogger) do  # Don't log anything from the writing of protobuffs since 🔁
    writeproto(...)
end
```

Maybe making a helper function that does that.

`NullLogger` is exported from the Logging stdlib

**Comment by PhilipVinc** [comment]
@JJMinton I am very busy and won't be able to address this soon, but if you want to open a PR implementing the fix I'll gladly review and merge it.

**Comment by PhilipVinc** [comment]
closed by #100
